### ISSUE TITLE
Rustfmt duplicates commas in struct declarations

### ISSUE THREAD

**Issue by xTibor** [rust-lang/rustfmt#4714]
**Describe the bug**

There's a weird interaction between comments and commas in struct declarations that leads to duplication of commas and breaking the syntax of the formatted code (`error: expected identifier, found ','`)

**To Reproduce**

Input:
```rust
struct Something {
    field: u32/*a*/,/*b*/
}
```

Formatting result:
```rust
struct Something {
    field: u32, /*a*/,/*b*/
}
```

**Expected behavior**

Do not duplicate commas.

**Meta**

- rustfmt version: 1.4.36-nightly (2021-02-07 7de6968)
- From where did you install rustfmt?: playground
- How do you run rustfmt: playground

**Comment by xTibor** [comment]
Just noticed there's a similar issue at #3876. My repro is in a different context than discussed there, however if the root cause is the same this issue can be closed as a dupe.

**Comment by whizsid** [comment]
@xTibor Yes it was a duplicate and #4520 will fix it.

**Comment by calebcartwright** [comment]
Thanks for the report @xTibor and @whizsid for pulling this in as a test case on #4520 :+1:

**Comment by calebcartwright** [comment]
Resolved via 74768b7f599278f72bc954b22d9ec214976eee50
